applications/tari_validator_node/src/p2p/services/template_manager/service.rs

```diff
@@ -142,15 +143,27 @@ impl TemplateManagerService {
     fn handle_completed_download(&mut self, download: DownloadResult) -> Result<(), TemplateManagerError> {
         match download.result {
             Ok(bytes) => {
-                // TODO: Validate binary against hash
                 info!(
                     target: LOG_TARGET,
                     "✅ Template {} downloaded successfully", download.template_address
                 );
+
+                // validation of the downloaded template binary hash
+                let actual_binary_sha = calculate_template_binary_hash(&bytes);
+                let template_status = if actual_binary_sha.to_vec() == download.expected_binary_sha.to_vec() {
```

Comment:
to_vec allocates on the heap which is unnecessary for the purposes of comparing the bytes
```suggestion
                let template_status = if actual_binary_hash.as_slice() == download.expected_binary_hash.as_slice() {
```